### PR TITLE
Update to project to Spring Framework 5.1.x and Spring Boot 2.1.x

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Comment by geoand** [comment]
@pavolloffay @objectiser any idea why [this](https://github.com/opentracing-contrib/java-spring-web/blob/611e66dd4c840b14f903addfd544b6d53a9ef1cd/opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java#L298) assertion is no longer working?

**Comment by objectiser** [comment]
@geoand No - what log events does it contain?

**Comment by geoand** [comment]
@objectiser see screenshot below:

screenshot from 2018-11-16 11-50-25

**Comment by geoand** [comment]
@objectiser see below:

selection_010

**Comment by geoand** [comment]
@pavolloffay Do you think I should just remove the assertion?
I agree with @objectiser that the log events seems reasonable

**Comment by pavolloffay** [comment]
Apparently, they have changed something in the processing of the request. It seems that interceptors are called requests without mapping.

I would add assertion on those events ` assertLogEvents(span.logEntries(), Arrays.asList("preHandle", "afterCompletion"));`

**Comment by geoand** [comment]
@pavolloffay thanks! I will update the PR

File: pom.xml [hunk 1]
```diff
@@ -45,10 +45,10 @@
     <version.io.opentracing.contrib-opentracing-tracerresolver>0.1.5</version.io.opentracing.contrib-opentracing-tracerresolver>
     <version.io.opentracing.contrib-opentracing-spring-tracer-configuration-starter>0.1.0</version.io.opentracing.contrib-opentracing-spring-tracer-configuration-starter>
     <version.junit>4.12</version.junit>
-    <version.org.mockito-mockito-core>1.10.19</version.org.mockito-mockito-core>
-    <version.org.springframework.boot>1.5.12.RELEASE</version.org.springframework.boot>
+    <version.org.mockito-mockito-core>2.23.4</version.org.mockito-mockito-core>
+    <version.org.springframework.boot>2.1.0.RELEASE</version.org.springframework.boot>
     <!-- Should match version from https://github.com/spring-projects/spring-boot/blob/1.5.x/spring-boot-dependencies/pom.xml  -->
```
Comment (pavolloffay) [inline pom.xml]: could you please also change this link poiting to SB 2.x?  https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-dependencies/pom.xml

Comment (geoand) [reply pom.xml]: Good catch!

**Comment by geoand** [comment]
@pavolloffay PR updated with your suggestions

**Comment by pavolloffay** [comment]
:+1: go ahead merge and release :)

**Comment by geoand** [comment]
Awesome! So version `1.0.0` then?

**Comment by pavolloffay** [comment]
oh sorry yes 1.x

**Pull request merged**